File: ios/Classes/UxtrackerFlutterSdkPlugin.swift
import Flutter
import UIKit
import UXTrackerSDK

@MainActor
public class UxtrackerFlutterSdkPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "uxtracker_flutter_sdk", binaryMessenger: registrar.messenger())
    let instance = UxtrackerFlutterSdkPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initialize":
        guard let args = call.arguments as? [String: Any],
              let apiKey = args["apiKey"] as? String,
              let setupDict = args["setup"] as? [String: Any],
              let flushInterval = setupDict["flushInterval"] as? Double,
              let batchSize = setupDict["batchSize"] as? Int else {
            result(FlutterError(code: "INVALID_ARGUMENTS", message: "Invalid arguments for initialize", details: nil))
            return
        }

        let uxTrackerSetup = UXTrackerSetup(flushInterval: flushInterval, batchSize: batchSize)
        UXTracker.shared.initialize(apiKey: apiKey, uxTrackerSetup: uxTrackerSetup)
        print("initialized completed")
        result(nil)
    case "track":
        if let args = call.arguments as? [String: Any],
               let event = args["event"] as? String,
               let properties = args["properties"] as? [String: String] {
                
                UXTracker.shared.track(eventName: event, userProperties: properties)
                
                result(nil)
            } else {
                result(FlutterError(code: "INVALID_ARGUMENTS", message: "Missing or invalid arguments for 'track'", details: nil))
            }
        print("event sent")
        result(nil)
    case "identify":
        if let args = call.arguments as? [String: Any],
           let userId = args["userId"] as? String {
            
            UXTracker.shared.identify(userId: userId)
            
            result(nil)
        } else {
            result(FlutterError(code: "INVALID_ARGUMENTS", message: "Missing or invalid arguments for 'identify'", details: nil))
        }
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: android/src/main/kotlin/com/wzagroup/uxtrackersdkflutter/uxtracker_flutter_sdk/UxtrackerFlutterSdkPlugin.kt
package com.wzagroup.uxtrackersdkflutter.uxtracker_flutter_sdk

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import com.wzagroup.uxtracker_android_sdk.UXTracker
import com.wzagroup.uxtracker_android_sdk.models.UXTrackerSetup

/** UxtrackerFlutterSdkPlugin */
class UxtrackerFlutterSdkPlugin :
    FlutterPlugin,
    MethodCallHandler {
    // The MethodChannel that will the communication between Flutter and native Android
    //
    // This local reference serves to register the plugin with the Flutter Engine and unregister it
    // when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel
    private lateinit var context: Context

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "uxtracker_flutter_sdk")
        channel.setMethodCallHandler(this)
        context = flutterPluginBinding.applicationContext
    }

    override fun onMethodCall(
        call: MethodCall,
        result: Result
    ) {
        when (call.method) {
            "initialize" -> {
                val args = call.arguments as Map<*, *>
                val apiKey = args["apiKey"] as String

                val setupMap = args["setup"] as? Map<*, *>
                val flushInterval = (setupMap?.get("flushInterval") as? Long) ?: 40
                val batchSize = (setupMap?.get("batchSize") as? Int) ?: 20
                val setup = UXTrackerSetup(flushInterval, batchSize)

                UXTracker.shared().initialize(context, apiKey = apiKey, setup = setup)
                result.success(null)
            }

            "track" -> {
                val args = call.arguments as Map<*, *>
                val event = args["event"] as String
                val properties = args["properties"] as? Map<String, Any> ?: emptyMap()

                UXTracker.shared().track(event, properties)
                result.success(null)
            }

            "identify" -> {
                val args = call.arguments as Map<*, *>
                val userId = args["userId"] as String

                UXTracker.shared().identify(userId)
                result.success(null)
            }

            "reset" -> {
                UXTracker.shared().reset()
                result.success(null)
            }

            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
